File: backend/ProjectProgressManagementSystem/Models/WeekData.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using ProjectProgressManagementSystem.Utilities;

namespace ProjectProgressManagementSystem.Models
{
    [PrimaryKey(nameof(UserId), nameof(ProjectId), nameof(Year), nameof(Month))]
    public class WeekData
    {
        [Key, Column(Order = 1)]
        [YearRange(2000)]
        public int Year { get; set; } = DateTime.Now.Year;

        [Key, Column(Order = 2)]
        [Range(1, 12)]
        public int Month { get; set; } = DateTime.Now.Month;

        [Key, Column(Order = 3)]
        public int UserId { get; set; }
        public User User { get; set; }

        [Key, Column(Order = 4)]
        public int ProjectId { get; set; }
        public Project Project { get; set; }

        public int Week1 { get; set; }
        public int Week2 { get; set; }
        public int Week3 { get; set; }
        public int Week4 { get; set; }
        public int? Week5 { get; set; }
    }
}

File: backend/app/models/week_data.py
from sqlalchemy import ForeignKey, Integer
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db.base import Base


class WeekData(Base):
    __tablename__ = "WeekData"

    year: Mapped[int] = mapped_column(Integer, primary_key=True, nullable=False)
    month: Mapped[int] = mapped_column(Integer, primary_key=True, nullable=False)
    user_id: Mapped[int] = mapped_column(Integer, ForeignKey("User.id"), primary_key=True, nullable=False)
    project_id: Mapped[int] = mapped_column(Integer, ForeignKey("Project.id"), primary_key=True, nullable=False)

    week1: Mapped[int] = mapped_column(Integer, default=0, nullable=False)
    week2: Mapped[int] = mapped_column(Integer, default=0, nullable=False)
    week3: Mapped[int] = mapped_column(Integer, default=0, nullable=False)
    week4: Mapped[int] = mapped_column(Integer, default=0, nullable=False)
    week5: Mapped[int | None] = mapped_column(Integer, nullable=True)

    user: Mapped["User"] = relationship(back_populates="week_data")  # noqa: F821
    project: Mapped["Project"] = relationship(back_populates="week_data")  # noqa: F821
